src/pageObjects/components/SignUpForm.js

```diff
@@ -0,0 +1,74 @@
+import BaseComponent from "../BaseComponent.js";
+import {expect} from "@playwright/test";
+
+export default class SignUpForm extends BaseComponent {
+
+    _inputName = '#signupName';
+    _inputLastName = '#signupLastName';
+    _inputEmail = '#signupEmail';
+    _inputPassword = '#signupPassword';
+    _inputRepeatPassword = '#signupRepeatPassword';
+    _registerButton = '.modal-footer button';
+    _alertDanger = '.alert-danger';
+    _errors = '.invalid-feedback';
+
+    constructor(page, inputValues) {
+        super(page, page.locator('div.modal-content'));
+        this.inputValues = inputValues;
+        this.inputName = this._container.locator(this._inputName);
+        this.inputLastName = this._container.locator(this._inputLastName);
+        this.inputEmail = this._container.locator(this._inputEmail);
+        this.inputPassword = this._container.locator(this._inputPassword);
+        this.inputRepeatPassword = this._container.locator(this._inputRepeatPassword);
+        this.registerButton = this._container.locator(this._registerButton);
+        this.alertDanger = this._container.locator(this._alertDanger, {hasText: 'User already exists'});
+        this.errors = this._container.locator(this._errors);
+    }
+
+    #inputs = () => {
+        return {
+            signupName: this.inputName,
+            signupLastName: this.inputLastName,
+            signupEmail: this.inputEmail,
+            signupPassword: this.inputPassword,
+            signupRepeatPassword: this.inputRepeatPassword
+        }
+    }
+
+    async registerNewUser(isNegative = false, data = this.inputValues) {
```

Comment:
`registerNewUser` should perform positive scenario

What I would suggest 

1. Create simple method `fill`. It should only fill form and should not care about data is positive or negative. It just fills form
2. `registerNewUser` invokes` fill` and does `this.this.registerButton.click()`. Also you can add assertions if registration was successful
3.About negative scenarios - create a method for validation inputs and error messages. Call `fill` from test and then cal this validation method.

The main take - make your methods more atomic. Read about Single responsibility principle. Each methods should have only one responsibility (fill, check errors etc)